test/visual/CMakeLists.txt

```diff
@@ -7,11 +7,17 @@ add_executable(mapnik-test-visual
 target_link_libraries(mapnik-test-visual PRIVATE
     Catch2::Catch2
     Boost::program_options
-    Boost::filesystem
     mapnik::mapnik
     mapnik::agg
     ICU::data ICU::i18n ICU::uc # needed for the static build (TODO: why isn't this correctly propagated from mapnik::mapnik?)
 )
+
+if(CMAKE_CXX_STANDARD LESS 17 OR USE_BOOST_FILESYSTEM)
+target_link_libraries(mapnik-test-visual PRIVATE
+    Boost::filesystem
+)
+endif()
+
```

Comment:
```suggestion
```
These lines shouldn't be needed, since `mapnik::mapnik` either depends on filesystem or not.